paris-processor/src/main/java/com/airbnb/paris/processor/writers/KotlinStyleExtensionsFile.kt

```diff
@@ -0,0 +1,129 @@
+package com.airbnb.paris.processor.writers
+
+import com.airbnb.paris.processor.STYLE_CLASS_NAME
+import com.airbnb.paris.processor.framework.*
+import com.airbnb.paris.processor.framework.AndroidClassNames.ATTRIBUTE_SET
+import com.airbnb.paris.processor.framework.AndroidClassNames.STYLE_RES
+import com.airbnb.paris.processor.models.BaseStyleableInfo
+import com.airbnb.paris.processor.models.StyleableInfo
+import com.squareup.kotlinpoet.*
+
+/**
+ * This generates a kt file with extension functions to style a specific view.
+ *
+ * For example, for ImageView this will generate a "ImageViewStyleExtensions.kt" file, with several
+ * extensions on ImageView to apply a style.
+ *
+ * These extensions achieve the same effect as using the Paris class.
+ */
+internal class KotlinStyleExtensionsFile(
+    packageName: String,
+    private val styleables: List<BaseStyleableInfo>
+) : SkyKotlinFile(packageName, "ParisExtensions", {
+
+    styleables.forEach { styleable ->
+
+        val builderParam = ParameterSpec.builder(
+            "builder",
+            LambdaTypeName.get(
+                receiver = styleable.styleBuilderClassName.toKPoet(),
+                returnType = UNIT
+            )
+        ).build()
+
+
+        /**
+         * An extension for setting a Style object on the view.
+         *
+         * Usage is like: "view.style(styleObject)"
```

Comment:
Where did we land on `.style(...)` vs `.applyStyle(...)`? The latter seems a bit more consistent with `.applyGreenStyle()` and in the future quite possibly with `.applyStyle { /* builder functions */ }`.